catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -549,9 +566,11 @@ private void addFileToQueue(ArrayBlockingQueue<Metacard> metacardQueue, long sta
 
         if (result != null) {
             try {
-                metacardQueue.put(result);
                 phaser.register();
+                metacardQueue.put(result);
```

Comment:
Registering (counting up) first and then counting it back down should an exception be thrown.